src/cache.js

```diff
@@ -9,16 +10,27 @@ const PLATFORM = os.platform();
 const CACHE_PATHS = [path.join(HOME, ".foundry/cache/rpc")];
 
 async function restoreRPCCache() {
-  const key = PLATFORM + "-foundry-chain-fork-" + github.context.sha;
+  const primaryKey = PLATFORM + "-foundry-chain-fork-" + github.context.sha;
   const restoreKeys = [PLATFORM + "-foundry-chain-fork-"];
-  await cache.restoreCache(CACHE_PATHS, key, restoreKeys);
+  const cacheKey = await cache.restoreCache(CACHE_PATHS, primaryKey, restoreKeys);
+  if (!cacheKey) {
+    core.info("Cache not found");
+    return;
+  }
+  core.info(`Cache restored from key: ${cacheKey}`);
 }
 
 async function saveCache() {
-  const key = PLATFORM + "-foundry-chain-fork-" + github.context.sha;
-  if (fs.existsSync(CACHE_PATHS[0])) {
-    await cache.saveCache(CACHE_PATHS, key);
+  const primaryKey = PLATFORM + "-foundry-chain-fork-" + github.context.sha;
+  if (!fs.existsSync(CACHE_PATHS[0])) {
+    core.info(`Cache path does not exist, not saving cache : ${CACHE_PATHS[0]}`);
+    return;
+  }
+  const cacheId = await cache.saveCache(CACHE_PATHS, primaryKey);
+  if (cacheId === -1) {
+    return;
```

Comment:
when does this happen? should we add a log?

Reply:
Looks like it happens when there is an existing cache with the primary key thus fails saving the cache. 
It doesn't throw an error and prints a message like this before returning with -1.

```
Failed to save: Unable to reserve cache with key linux-foundry-chain-fork-c2ec5babbad85fecbae9c860472722f98f36c164, another job may be creating this cache. More details: Cache already exists. Scope: refs/heads/master, Key: linux-foundry-chain-fork-c2ec5babbad85fecbae9c860472722f98f36c164, Version: d8cfedd31e23b815ad861b2e68523a31664eaf28c2ddc7b682eb115646a473d9
```

Reply:
https://github.com/actions/toolkit/blob/415c42d27ca2a24f3801dd9406344aaea00b7866/packages/cache/src/cache.ts#L245-L246